src/core/compiler.jl

```diff
@@ -59,6 +70,22 @@ function vsym(expr::Union{Expr, Symbol})
     throw("VarName: Mis-formed variable name $(expr)!")
 end
 
+macro vinds(expr::Union{Expr, Symbol})
+    expr |> vinds |> esc
+end
+function vinds(expr::Union{Expr, Symbol})
+    ex = deepcopy(expr)
+    inds = Expr(:tuple)
+    (ex isa Symbol) && return inds
+    (ex.head == :ref) || throw("VarName: Mis-formed variable name $(expr)!")
+    while ex.head == :ref
```

Comment:
Is this guaranteed to terminate?

Reply:
Hmm, good question, it will either terminate or error eventually, assuming the input expression is finite which it is.